aten/src/ATen/native/AveragePool3d.cpp

```diff
@@ -396,15 +399,16 @@ void avg_pool3d_backward_out_frame(
             }
 
             /* scatter gradients out to footprint: */
-            scalar_t val  = *op++;
+            accscalar_t val  = static_cast<accscalar_t>(*op++);
+            accscalar_t val_scaled = val / divide_factor;
```

Comment:
This value is const and applied to all values in the for loop below, should be statically cast once to scalar_t and applied. Casting in the loop is inefficient.